src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIContainerService.java

```diff
@@ -366,17 +359,20 @@ public Path createBuildScript(ProgrammingExerciseParticipation participation) {
                 cd /testing-dir
                 """);
 
-        if (actions != null) {
-            actions.forEach(action -> buildScript.append(action.getScript()).append("\n"));
-        }
-        else {
+        actions.forEach(action -> buildScript.append(action.getScript()).append("\n"));
+
+        // Fall back to hardcoded scripts for old exercises without windfile
+        // *****************
+        // TODO: delete
```

Comment:
TODO: delete?

Reply:
This will be deleted in the future. Currently still there, so existing exercises do not break.